react-native-reanimated.d.ts

```diff
@@ -238,8 +238,10 @@ declare module 'react-native-reanimated' {
       getNode(): ReactNativeScrollView;
     }
     export class Code extends Component<CodeProps> {}
-    export function createAnimatedComponent<S extends object, P extends { style?: StyleProp<S>; }>(component: ComponentType<P>): ComponentType<AnimateProps<S, P>>;
-
+    export function createAnimatedComponent<C extends ComponentClass, S extends object>(
+      component: C
+    ): ComponentType<AnimateProps<S, ComponentProps<C>>>;
```

Comment:
1. Why do we have a generic parameter for component here? I'm not quite sure how it works.
2. `S extends object` – if we want to allow only dict-like values we should use `S extends Record<string, unknown>`. We can change it in the AnimateProps generic arg too.

Reply:
Yes I would really like to fix this too. Agree for 2. I'm also wondering if it would be possible to rewrite AnimatedProps to be in the Shape AnimatedProps<ProType> only. That would solve all our problems.

Reply:
The problem with Record is that it doesn't accepts interfaces. I would like to remove S from AnimateProps but that seems a bit tricky.

Reply:
> The problem with Record is that it doesn't accepts interfaces.

Could you give an example?

Reply:
an example is in useAnimatedGestureHandler where TContext needs to be using the type keyword and not the interface one.